### DISCUSSION TITLE
HTTP 2 doesn't work. Response http_method still indicates it was used.

### DISCUSSION THREAD

**Discussion by RafaelDiasCampos** [encode/httpx#2822] [Potential Issue] [answered]
I'm having an issue when using httpx with the http2=True option.
When I make a request, the value http_version shows that HTTP 2 was used, but if I inspect the packet using Wireshark, I can see that HTTP 1.1 was used instead.

Code used for testing:
```
import httpx

client = httpx.Client(http2=True)

r = client.get("https://www.google.com")

print(f"HTTP Version: {r.http_version}")
print(f"URL: {r.url}")
print(f"IP: {r.stream._stream._httpcore_stream._stream._connection._network_stream._sock.getpeername()[0]}")
```

Output suggests that HTTP 2 was used:

![Screenshot 2023-08-19 091822](https://github.com/encode/httpx/assets/66502547/04967abb-9d13-46d1-b7f7-2e9186f9b467)

But inspecting the packet with Wireshark shows version 1.1:

![Screenshot 2023-08-19 090826](https://github.com/encode/httpx/assets/66502547/8ef7b12c-5574-43d6-be34-1c2492ea8f4e)

Using httpx[http2] version 0.24.1. Tested both in Windows and Linux, behaviour was consistent in both OS.
Am I doing something wrong? I haven't been able to get HTTP 2 to work at all.

**Comment by karpetrosyan** [comment]
Hi Rafael! Are you certain that these are http/1.1 packets in wireshark? Also, if you just want to use http2, disable http1.1 by giving http1=False to the client.

**Comment by RafaelDiasCampos** [comment]
Oops, my bad. It seems it was HTTP 2 traffic. Thank you for the help.

**Comment by karpetrosyan** [comment, marked as answer]
> Oops, my bad. It seems it was HTTP 2 traffic. Thank you for the help.

If you thought it was the http/1.1 packet based on the 'http/1.1' in the packet bytes, I guess it's just the [tls alpn negotiation](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation), which is used to determine whether the server prefers http/1.1 or h2.
